resources/web/wwi/MouseEvents.js

```diff
@@ -210,41 +317,86 @@ class MouseEvents { // eslint-disable-line no-unused-vars
 
       this.moveParams.dx = x - this.state.initialX;
       this.moveParams.dy = y - this.state.initialY;
-      this.scene.viewpoint.translate(this.moveParams);
+
+      let targetRight = -scaleFactor * this.moveParams.dx;
+      let targetUp = scaleFactor * this.moveParams.dy;
+      let upVec = up(orientation);
+      let rightVec = right(orientation);
+      let targetR = rightVec.mul(targetRight);
+      let targetU = upVec.mul(targetUp);
+      let target = targetR.add(targetU);
+      WbWorld.instance.viewpoint.position = position.add(target);
+      WbWorld.instance.viewpoint.updatePosition();
+      this.scene.render();
     } else {
-      var touch1 = event.targetTouches['1'];
-      var x1 = Math.round(touch1.clientX);
-      var y1 = Math.round(touch1.clientY);
-      var distanceX = x - x1;
-      var distanceY = y - y1;
-      var newTouchDistance = distanceX * distanceX + distanceY * distanceY;
-      var pinchSize = this.state.touchDistance - newTouchDistance;
-
-      var moveX1 = x - this.state.x;
-      var moveX2 = x1 - this.state.x1;
-      var moveY1 = y - this.state.y;
-      var moveY2 = y1 - this.state.y1;
-      var ratio = window.devicePixelRatio || 1;
```

Comment:
We had issues with retina displays due to the `window.devicePixelRatio`.
If possible we should test this PR also on retina displays.